command/operator_debug_test.go

```diff
@@ -275,136 +183,152 @@ func TestDebug_ClientToServer(t *testing.T) {
 	srv, _, url := testServer(t, false, nil)
 	defer srv.Shutdown()
+
+	// Wait for leadership to establish
 	testutil.WaitForLeader(t, srv.Agent.RPC)
 
+	cases := []string{
+		"aws/ebs",
+		"gcp-*-1",
+	}
+	for _, pluginName := range cases {
+		cleanup := state.CreateTestCSIPlugin(srv.Agent.Server().State(), pluginName)
+		defer cleanup()
+	}
+
+	// Setup mock UI
 	ui := cli.NewMockUi()
 	cmd := &OperatorDebugCommand{Meta: Meta{Ui: ui}}
 
-	// Fails incorrect args
-	code := cmd.Run([]string{"some", "bad", "args"})
-	require.Equal(t, 1, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
-
-	// Fails illegal node ids
-	code = cmd.Run([]string{"-node-id", "foo:bar"})
-	require.Equal(t, 1, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
-
-	// Fails missing node ids
-	code = cmd.Run([]string{"-node-id", "abc,def", "-duration", "250ms"})
-	require.Equal(t, 1, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
-
-	// Fails bad durations
-	code = cmd.Run([]string{"-duration", "foo"})
-	require.Equal(t, 1, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	// Debug on the leader and all client nodes
+	code := cmd.Run([]string{"-address", url, "-duration", "250ms", "-server-id", "leader", "-node-id", "all", "-output", os.TempDir()})
+	assert.Equal(t, 0, code)
 
-	// Fails bad durations
-	code = cmd.Run([]string{"-interval", "bar"})
-	require.Equal(t, 1, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	// Bad plugin name should be escaped before it reaches the sandbox test
+	require.NotContains(t, ui.ErrorWriter.String(), "file path escapes capture directory")
+	require.Contains(t, ui.OutputWriter.String(), "Starting debugger")
 
-	// Fails existing output
-	format := "2006-01-02-150405Z"
-	stamped := "nomad-debug-" + time.Now().UTC().Format(format)
-	path := filepath.Join(os.TempDir(), stamped)
-	os.MkdirAll(path, 0755)
+	path := cmd.collectDir
 	defer os.Remove(path)
-	// short duration to prevent timeout
-	code = cmd.Run([]string{"-output", os.TempDir(), "-duration", "50ms"})
-	require.Equal(t, 2, code)
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
 
-	// Fails bad address
-	code = cmd.Run([]string{"-address", url + "bogus"})
-	assert.Equal(t, 1, code) // take note of failed return code, but continue to see why in the OutputWriter
-	require.NotContains(t, ui.OutputWriter.String(), "Starting debugger")
-	require.Contains(t, ui.ErrorWriter.String(), "invalid address")
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	var pluginFiles []string
+	for _, pluginName := range cases {
+		pluginFile := fmt.Sprintf("csi-plugin-id-%s.json", helper.CleanFilename(pluginName, "_"))
+		pluginFile = filepath.Join(path, "nomad", "0000", pluginFile)
+		pluginFiles = append(pluginFiles, pluginFile)
+	}
+
+	testutil.WaitForFiles(t, pluginFiles)
 }
 
-func TestDebugCapturedFiles(t *testing.T) {
+func TestDebug_CapturedFiles(t *testing.T) {
 	// NOTE: pprof tracing/profiling cannot be run in parallel
```

Comment:
Are there any tests in this file that this comment doesn't apply to? Maybe it should be a top-level comment?

Reply:
This is the only test which specifies an output directory, and looks for specific capture files to be present.